samples/compression/lz4 failed with arcmwdt toolchian

**Describe the bug**
samples/compression/lz4 failed with arcmwdt toolchian

**To Reproduce**
Steps to reproduce the behavior:
1. west build -b nsim_em -d build samples/compression/lz4
2. See error

**Logs and console output**
```
west build -b nsim_em -d build samples/compression/lz4
-- west build: generating a build system
Including boilerplate (Zephyr base): /SCRATCH2/Watson/zephyrproject/zephyr/cmake/app/boilerplate.cmake
-- Application: /SCRATCH2/Watson/zephyrproject/zephyr/samples/compression/lz4
-- Zephyr version: 2.6.99 (/SCRATCH2/Watson/zephyrproject/zephyr), build: zephyr-v2.6.0-474-gb4faf7fe6320
-- Found Python3: /depot/Python/Python-3.7.0/bin/python3.7 (found suitable exact version "3.7.0") found components: Interpreter 
-- Found west (found suitable version "0.11.0", minimum required is "0.7.1")
-- Board: nsim_em
-- Cache files will be written to: /u/zhiwei/.cache/zephyr
-- Using toolchain: zephyr 0.12.3 (/SCRATCH2/svc-arcoss_auto/tools/zephyr-sdk/0.12.3)
-- Found dtc: /SCRATCH2/svc-arcoss_auto/tools/zephyr-sdk/0.12.3/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.5.0", minimum required is "1.4.6")
-- Found BOARD.dts: /SCRATCH2/Watson/zephyrproject/zephyr/boards/arc/nsim/nsim_em.dts
-- Generated zephyr.dts: /SCRATCH2/Watson/zephyrproject/zephyr/build/zephyr/zephyr.dts
-- Generated devicetree_unfixed.h: /SCRATCH2/Watson/zephyrproject/zephyr/build/zephyr/include/generated/devicetree_unfixed.h
-- Generated device_extern.h: /SCRATCH2/Watson/zephyrproject/zephyr/build/zephyr/include/generated/device_extern.h
Parsing /SCRATCH2/Watson/zephyrproject/zephyr/Kconfig
Loaded configuration '/SCRATCH2/Watson/zephyrproject/zephyr/boards/arc/nsim/nsim_em_defconfig'
Merged configuration '/SCRATCH2/Watson/zephyrproject/zephyr/samples/compression/lz4/prj.conf'
Configuration saved to '/SCRATCH2/Watson/zephyrproject/zephyr/build/zephyr/.config'
Kconfig header saved to '/SCRATCH2/Watson/zephyrproject/zephyr/build/zephyr/include/generated/autoconf.h'
-- The C compiler identification is Clang 10.0.1
-- The CXX compiler identification is Clang 10.0.1
-- The ASM compiler identification is unknown
-- Found assembler: /global/apps/mwdt_2021.03/MetaWare/arc/bin/ccac
-- Warning: Did not find file Compiler/-ASM
CMake Warning at ../../../cmake/linker/arcmwdt/target.cmake:137 (message):
  MWDT toolchain does not support

             CONFIG_LINKER_ORPHAN_SECTION_WARN
Call Stack (most recent call first):
  ../../../CMakeLists.txt:246 (toolchain_ld_baremetal)


-- Configuring done
-- Generating done
-- Build files have been written to: /SCRATCH2/Watson/zephyrproject/zephyr/build
-- west build: building application
[1/110] Preparing syscall dependency handling

[7/110] Building C object zephyr/CMakeFiles/offsets.dir/arch/arc/core/offsets/offsets.c.obj
FAILED: zephyr/CMakeFiles/offsets.dir/arch/arc/core/offsets/offsets.c.obj 
/global/apps/mwdt_2021.03/MetaWare/arc/bin/ccac -DBUILD_VERSION=zephyr-v2.6.0-474-gb4faf7fe6320 -DKERNEL -D__LINUX_ERRNO_EXTENSIONS__ -D__ZEPHYR__=1 -I../kernel/include -I../arch/arc/include -I../include -Izephyr/include/generated -I../soc/arc/snps_nsim -I../lib/libc/newlib/include -I/SCRATCH2/Watson/zephyrproject/modules/lib/lz4/lib -Os -imacros /SCRATCH2/Watson/zephyrproject/zephyr/build/zephyr/include/generated/autoconf.h -Hnocrt -fno-common -g -Hnocopyr -HL -Hnosdata -Hoff=Stackcheck_alloca -imacros /SCRATCH2/Watson/zephyrproject/zephyr/include/toolchain/zephyr_stdint.h -Wformat -Wformat-security -Wno-format-zero-length -Wno-main-return-type -Wno-unaligned-pointer-conversion -Wno-incompatible-pointer-types-discards-qualifiers -Wno-typedef-redefinition -Wno-pointer-sign -Wpointer-arith -Wno-sometimes-uninitialized -Wno-shift-overflow -Wno-missing-braces -Wno-self-assign -Wno-address-of-packed-member -Wno-unused-function -Wno-initializer-overrides -Wno-section -Wno-unknown-warning-option -Wno-unused-variable -Wno-format-invalid-specifier -Wno-gnu -Wno-tautological-compare -Werror=implicit-int -ffunction-sections -fdata-sections -arcv2em -core3 -Xdiv_rem=radix2 -Xmpy_option=mpyd -Xbitscan -Xswap -Xbarrel_shifter -Xshift_assist -Xdsp2 -Xdsp_complex -Xdsp_divsqrt=radix2 -Xdsp_itu -Xdsp_accshift=full -Xfpus_div -Xfpu_mac -Xfpuda -Xfpus_mpy_slow -Xfpus_div_slow -Xbitstream -Xtimer0 -Xtimer1 -std=gnu99 -MD -MT zephyr/CMakeFiles/offsets.dir/arch/arc/core/offsets/offsets.c.obj -MF zephyr/CMakeFiles/offsets.dir/arch/arc/core/offsets/offsets.c.obj.d -o zephyr/CMakeFiles/offsets.dir/arch/arc/core/offsets/offsets.c.obj   -c /SCRATCH2/Watson/zephyrproject/zephyr/arch/arc/core/offsets/offsets.c
In file included from /SCRATCH2/Watson/zephyrproject/zephyr/arch/arc/core/offsets/offsets.c:25:
In file included from ../include/kernel.h:17:
In file included from ../include/kernel_includes.h:17:
In file included from ../include/zephyr/types.h:11:
../lib/libc/newlib/include/stdint.h:15:10: fatal error: 'newlib.h' file not found
#include <newlib.h>
         ^~~~~~~~~~
1 error generated.
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /depot/Python/Python-3.7.0/bin/cmake --build /SCRATCH2/Watson/zephyrproject/zephyr/build
```

**Environment (please complete the following information):**
 - OS: (Linux)
 - Toolchain (arcmwdt)
 - Commit SHA: b4faf7fe63201ad26b9c4ddf4742ad750c9f2d7a


**Additional context**
samples/compression/lz4 need `CONFIG_NEWLIB_LIBC=y`, while arcmwdt toolchain not support newlib, 
we need to add a filter in sample.yaml: `filter: TOOLCHAIN_HAS_NEWLIB == 1`

--- a/samples/compression/lz4/sample.yaml
+++ b/samples/compression/lz4/sample.yaml
@@ -4,6 +4,7 @@ sample:
 common:
   tags: compression lz4
   min_ram: 32
+  filter: TOOLCHAIN_HAS_NEWLIB == 1
   harness: console
   harness_config:
     type: one_line

--- a/tests/kernel/mem_protect/mem_protect/src/mem_partition.c
+++ b/tests/kernel/mem_protect/mem_protect/src/mem_partition.c
@@ -7,9 +7,12 @@
 #include <kernel.h>
 #include "mem_protect.h"
 
-static K_APP_DMEM(ztest_mem_partition) int var = 1356;
-static K_APP_BMEM(ztest_mem_partition) int zeroed_var = 20420;
-static K_APP_BMEM(ztest_mem_partition) int bss_var;
+/* Add volatile to disable pre-calculation in compile stage in some
+ * toolchain, such as arcmwdt toolchain.
+ */
+static volatile K_APP_DMEM(ztest_mem_partition) int var = 1356;
+static volatile K_APP_BMEM(ztest_mem_partition) int zeroed_var = 20420;
+static volatile K_APP_BMEM(ztest_mem_partition) int bss_var;
 
 /**
  * @brief Test assigning global data and BSS variables to memory partitions

